internal/engine/build_and_deployer_test.go

```diff
@@ -728,16 +579,14 @@ func TestDockerComposeLiveUpdate(t *testing.T) {
 	assert.Equal(t, 0, f.docker.PushCount)
 	assert.Equal(t, 1, f.docker.CopyCount)
 	assert.Equal(t, 1, len(f.docker.ExecCalls))
-	assert.Equal(t, 0, f.sCli.UpdateContainerCount,
-		"Expected no synclet UpdateContainer call, actual: %d", f.sCli.UpdateContainerCount)
-	if strings.Contains(f.k8s.Yaml, sidecar.DefaultSyncletImageName) {
-		t.Errorf("Should not deploy the synclet for a docker-compose build: %s", f.k8s.Yaml)
-	}
+	assert.Empty(t, f.k8s.Yaml, "expect no k8s YAML for DockerCompose resource")
+	assert.Empty(t, 0, f.k8s.ExecCalls,
+		"Expected no k8s Exec calls, actual: %d", f.k8s.ExecCalls)
 	f.assertContainerRestarts(1)
 }
 
 func TestReturnLastUnexpectedError(t *testing.T) {
-	f := newBDFixture(t, k8s.EnvGKE, container.RuntimeDocker)
+	f := newBDFixture(t, k8s.EnvDockerDesktop, container.RuntimeDocker)
```

Comment:
We should probably fix the SanchoLiveUpdate manifest (i.e. the one we use by default in most of these tests) to NOT have a restart_container step in it, since that's been deprecated on the user side, and having that step in the manifest means we can't test it against UpdateModeKubectlExec (need to use a different update mode or a different test manifest). [Tracked that work here](https://app.clubhouse.io/windmill/story/8279/not-blocking-rm-restart-container-from-our-default-test-manifests) but for now it's easier to just change the k8s cluster so we get an update mode that allows `restart_container()` -- we'll deal with the rest if/when we actually deprecate that codepath.